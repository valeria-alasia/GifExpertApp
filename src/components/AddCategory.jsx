import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {

        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const value = inputValue.trim();
        if (value.length <= 1) return;
        onNewCategory(value)
        setInputValue('')
    } 

    return (

        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type='text'
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}/>
        
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}