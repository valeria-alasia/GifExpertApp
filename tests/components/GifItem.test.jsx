import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {
  const title = 'Saitama';
  const url = 'https://one-punch.con/saitama.jpg';
//   const id = '123';

  test('debe hacer match con el snapshot', () => {
   
    const {container} = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar la imagen con el URL y el ALT indicados', () => {

    render(<GifItem title={title} url={url} />);
    //screen.debug();
   
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

  });

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();

  });

})