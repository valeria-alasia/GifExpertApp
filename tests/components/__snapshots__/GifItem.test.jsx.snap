// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.con/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
